[DOCS] Adds example of semantic search with ELSER (#95992)
elastic/elasticsearch: docs/reference/search/search-your-data/semantic-search-elser.asciidoc

that the model created based on your text - must be created.  The destination 
index must have a field with the <<rank-features, `rank_features`>> field type 
to index the ELSER output.

[source,console]
----
PUT my-index
{
  "mappings": {
    "properties": {
      "ml.tokens": {
        "type": "rank_features" <1>
      },
      "text_field": {
        "type": "text" <2>
      }
    }
  }
}
----
// TEST[skip:TBD]
<1> The field that contains the prediction is a `rank_features` field.
<2> The text field from which to create the sparse vector representation.

[discrete]
[[inference-ingest-pipeline]]
=== Create an ingest pipeline with an inference processor

Create an <<ingest,ingest pipeline>> with an 
<<inference-processor,{infer} processor>> to use ELSER to infer against the data 
that is being ingested in the pipeline.

[source,console]
----
PUT _ingest/pipeline/elser-v1-test
{
  "processors": [
    {
      "inference": {
        "model_id": ".elser_model_1",
        "target_field": "ml",
        "field_map": {
          "text": "text_field"
        },
        "inference_config": {
          "text_expansion": { <1>

>             "results_field": "ml.tokens"

Author: ```suggestion
            "results_field": "tokens"
```

      }
    }
  }
}
----
// TEST[skip:TBD]
<1> The field that contains the prediction is a `rank_features` field.
<2> The text field from which to create the sparse vector representation.

[discrete]
[[inference-ingest-pipeline]]
=== Create an ingest pipeline with an inference processor

Create an <<ingest,ingest pipeline>> with an 
<<inference-processor,{infer} processor>> to use ELSER to infer against the data 
that is being ingested in the pipeline.

[source,console]
----
PUT _ingest/pipeline/elser-v1-test
{
  "processors": [
    {
      "inference": {
        "model_id": ".elser_model_1",
        "target_field": "ml",
        "field_map": {
          "text": "text_field"
        },
        "inference_config": {
          "text_expansion": { <1>
            "results_field": "ml.tokens"
          }
        }
      }
    }
  ]
}
----
// TEST[skip:TBD]
<1> The `text_expansion` inference type needs to be used in the {infer} ingest 
processor.

[discrete]
[[load-data]]
=== Load data

In this step, you load the data that you later use in the {infer} ingest 
pipeline to extract tokens from it.

NOTE: Only the first 512 extracted tokens of the ingested docs are considered 

> during semantic search.

Reviewer: ```suggestion
during semantic search with ELSER v1.
```

> during semantic search.

Reviewer: ```suggestion
during semantic search with ELSER v1.
```

Download the file and upload it to your cluster using the 
{kibana-ref}/connect-to-elasticsearch.html#upload-data-kibana[Data Visualizer] 
in the {ml-app} UI. Assign the name `id` to the first column and `text` to the 
second column. The index name is `test-data`. Once the upload is complete, you 
can see an index named `test-data` with 182469 documents.

[discrete]
[[reindexing-data-elser]]
=== Ingest the data through the {infer} ingest pipeline

Create the tokens from the text by reindexing the data throught the {infer} 
pipeline that uses ELSER as the inference model.

[source,console]
----
POST _reindex?wait_for_completion=false
{
  "source": {
    "index": "test-data"
  },
  "dest": {
    "index": "my-index",
    "pipeline": "elser-v1-test"
  }
}
----
// TEST[skip:TBD]

The call returns a task ID to monitor the progress:

[source,console]
----
GET _tasks/<task_id>
----
// TEST[skip:TBD]

You can also open the Trained Models UI, select the Pipelines tab under ELSER to 
follow the progress. It may take a couple of minutes to complete the process.

[discrete]
[[text-expansion-query]]
=== Semantic search by using the `text_expansion` query

> To perform semantic search, pass a `text_expansion` object to the search API 
> call, and provide the query text and the ELSER model ID. The example below uses 

Reviewer 2: ```suggestion
To perform semantic search, use the `text_expansion` query, 
and provide the query text and the ELSER model ID. The example below uses 
```
